ios/brave-ios/Sources/Brave/Frontend/Browser/New Tab Page/Backgrounds/NewTabPageBackgroundView.swift

```diff
@@ -42,4 +59,32 @@ class NewTabPageBackgroundView: UIView {
   required init(coder: NSCoder) {
     fatalError()
   }
+
+  private func parseColorFromFilename(filename: String) -> UIColor {
+    var color: String?
+    if let range = filename.range(of: "\\.RGB[a-fA-F0-9]+\\.", options: .regularExpression) {
+      color = filename[range].replacingOccurrences(of: ".RGB", with: "")
+        .replacingOccurrences(of: ".", with: "")
+    }
+
+    guard let color = color,
+      color.count == 6
+    else {
+      return UIColor.black
+    }
+
+    var rgbValue: UInt64 = 0
+    Scanner(string: color).scanHexInt64(&rgbValue)
+
+    return UIColor(
+      red: CGFloat((rgbValue & 0xFF0000) >> 16) / 255.0,
+      green: CGFloat((rgbValue & 0x00FF00) >> 8) / 255.0,
+      blue: CGFloat(rgbValue & 0x0000FF) / 255.0,
+      alpha: CGFloat(1.0)
+    )
```

Comment:
@aseren Brandon I believe mentioned a helper we could use for this in a previous comment?

Reply:
The initial comment from Brandon and my reply: https://github.com/brave/brave-core/pull/22701#discussion_r1558324356
This is file private extension UIColor, we cannot use it.

Reply:
See `public convenience init(colorString: String) {` in `UIColorExtensions` in `Shared`.

Reply:
changed to `init(colorString: String)` as per comment from Terry